controller/pkg/usertokens/pkitokens/jwt.go

```diff
@@ -20,28 +17,30 @@ import (
 // for validating the tokens. The public key is provided out-of-band.
 type PKIJWTVerifier struct {
 	JWTCertPEM  []byte
-	jwtCert     *x509.Certificate
+	keys        []crypto.PublicKey
 	RedirectURL string
 }
 
 // NewVerifierFromFile assumes that the input is provided as file path.
 func NewVerifierFromFile(jwtcertPath string, redirectURI string, redirectOnFail, redirectOnNoToken bool) (*PKIJWTVerifier, error) {
 	jwtCertPEM, err := ioutil.ReadFile(jwtcertPath)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to read jwt signing certificate from file: %s", err)
+		return nil, fmt.Errorf("failed to read JWT signing certificates or public keys from file: %s", err)
 	}
 	return NewVerifierFromPEM(jwtCertPEM, redirectURI, redirectOnFail, redirectOnNoToken)
 }
 
 // NewVerifierFromPEM assumes that the input is a PEM byte array.
 func NewVerifierFromPEM(jwtCertPEM []byte, redirectURI string, redirectOnFail, redirectOnNoToken bool) (*PKIJWTVerifier, error) {
-	jwtCertificate, err := tglib.ParseCertificate(jwtCertPEM)
-	if err != nil {
-		return nil, fmt.Errorf("Failed to read jwt signing certificate from PEM: %s", err)
+	keys, err := parsePublicKeysFromPEM(jwtCertPEM)
+	// pay attention to the return format of parsePublicKeysFromPEM
+	// when checking for an error here
+	if keys == nil && err != nil {
+		return nil, fmt.Errorf("failed to read JWT signing certificates or public keys from PEM: %s", err)
 	}
 	return &PKIJWTVerifier{
 		JWTCertPEM:  jwtCertPEM,
-		jwtCert:     jwtCertificate,
+		keys:        keys,
```

Comment:
Detail. But we even need to hold the PEM file at this point?

Reply:
Yes, because it is needed in `NewVerifier()` which can reinitialize a verifier from the same struct which is needed in `policy.ToPrivatePolicy()`.